### ISSUE TITLE
zippedArchive not working

### ISSUE THREAD

**Issue by farahamawi** [winstonjs/winston-daily-rotate-file#155]
Hi there
I'm using winston@3.0.0-rc6, and daily rotate file, it's working fine with me, the problem is that daily rotate file is archiving my old log files but not zipped, 
ps: I'm setting zippedArchive to true.

here is apart of my configuration

  backend: createLogger({
    levels: { error: 0, info: 1 },
    format: combine(
      timestamp(),
      splat(),
      simple(),
      myFormat,
    ),
    exceptionHandlers: [
      new transports.DailyRotateFile({
        filename: 'logs/%DATE%-exceptions.log',
        maxSize: '20m',
        maxFiles: '1d',
      }),
      new transports.Console({
        level: 'error',
        colorize: true,
      }),
    ],
    transports: [
      new transports.DailyRotateFile({
        name: 'backend-logs',
        filename: 'logs/%DATE%-backend.log',
        zippedArchive: true,
        maxSize: '20m',
      }),
      new transports.Console({
        level: 'error',
        colorize: true,
      }),
    ],
    exitOnError: false,
  }),

the old file 'logs/%DATE%-backend.log' is exist but its not zipped

I upgrade the verion of daily rotate file to 2.0.0-beta but I got an error => common.log is not a function
Can anyone help me with this!!!!

**Comment by mattberther** [comment]
@farahamawi With the configuration above, the log file will only rotate once per day. And, also, if the process stops - for whatever reason - and you start up a new process on a different day, it will not archive historical files.

If none of these items apply to your situation, please provide the smallest piece of code that demonstrates the problem, and I'd be happy to look into it further.

**Comment by farahamawi** [comment]
Hi @mattberther thank you for replaying, as you can see I set the exception file to rotate once per day(which is fine with me) , but not the same for backend.logs 
new transports.DailyRotateFile({
name: 'backend-logs',
filename: 'logs/%DATE%-backend.log',
zippedArchive: true,
maxSize: '20m',
}),

FYI  my system is archiving old files but its not compressed(zipped) thats the problem that I'm looking to solve.

**Comment by mattberther** [comment]
Thank you, @farahamawi. Please also confirm the versions of the components you're using (winston, winston-daily-rotate-file, and nodejs).

**Comment by farahamawi** [comment]
winston: 3.0.0-rc5
winston-daily-rotate-file: 3.1.4
nodejs: 9.5.0

**Comment by farahamawi** [comment]
Update
winston: 3.0.0-rc5
winston-daily-rotate-file: 3.2.1
nodejs: 9.5.0

**Comment by farahamawi** [comment]
Hi Matt, your script is working fine on my system,  it creates 3 files, 2 of them gzipped, but I'm a little bit confused if we remove the for loop and restart the server 3 times its creating new files but not gzipped
could you explain why.

you can repreduce that by just removing the for loop

var winston = require('winston');
var randomstring = require('randomstring');
require('winston-daily-rotate-file');


var transports = [
    new winston.transports.DailyRotateFile({
        name: 'logs',
        filename: 'logs/%DATE%.log',
        maxSize: '1k',
        zippedArchive: true
    }),
    new winston.transports.Console({
        colorize: true
    })
];

var logger = winston.createLogger({
    transports: transports
});

logger.info(randomstring.generate(2056));

try restart the server 3 times 
Please confirm that its creating 3 files not gzipped and explain why, Thank you for your time

**Comment by farahamawi** [comment]
Got it, Thank you Matt

**Comment by vishalgarale** [comment]
@mattberther Is there is any way to handle this use case? As we are also facing same trouble sometimes we need to stop server hence logs files are not getting compressed when we start server again with "worker_id" appended to file name. 

Or only way we have is that put separate crontab for compression?

**Comment by mattberther** [comment]
Yes. The transport will not modify files that it was not responsible for generating during the current process. If it would, it would be a pretty glaring security hole.
